thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/text/TTextProtocol.java

```diff
@@ -717,6 +733,36 @@ private BaseContext getCurrentContext() {
         return contextStack.peek();
     }
 
+    /**
+     * Prepare the current parsing context for writing.
+     */
+    private void writeCurrentContext() {
+        getCurrentContext().write();
+        updateCurrentFieldName();
+    }
+
+    /**
+     * Prepare the current parsing context for reading.
+     */
+    private void readCurrentContext() {
+        getCurrentContext().read();
+        updateCurrentFieldName();
+    }
+
+    /**
+     * Update the current field name, if necessary.
+     *
+     * <p>After every read/write operation in {@link MapContext},
+     * the field name should toggle between the map's key and value.</p>
+     */
+    private void updateCurrentFieldName() {
+        if (getCurrentContext() instanceof MapContext) {
+            currentFieldName.pop();
+            final String suffix = getCurrentContext().isMapKey() ? MAP_KEY_SUFFIX : MAP_VALUE_SUFFIX;
+            currentFieldName.push(currentFieldName.peek() + suffix);
```

Comment:
How about just using an `ArrayList` and updating the last element?

Reply:
Thanks for the feedback!

In my opinion, using a `Stack` makes the code much cleaner, since we mostly care about the top 1 or 2 items.

This way, we don't have to worry about the suffix, with the exception of `updateCurrentFieldName()`. I was thinking of doing `endsWith()` on the last item here, with slicing and appending the new suffix, but it's a bit messier.

The current approach also handles the first nested map a bit nicer, since we don't need to setup the correct suffix when pushing to the stack (only handled by this method). This way, we only need to push the same field name again, the rest is handled in this method.

Example:
```
1. A
2. A-> A   # readMapBegin() L551
3. A-> A$k # readCurrentContext()
4. A-> A$v # readCurrentContext()
...
```

But if you think using an `ArrayList` is better, I can change it! I can also only add the `$k` prefix in `readMapBegin()` and `writeMapBegin()`, since it might make debugging easier and there's one "edge case" less.

Reply:
I think it's OK as it is. Please feel free to choose what's right for you.

Reply:
Thanks. In this case, I'd prefer keeping `Stack` since the code is cleaner. 🙇‍♂️